editoast/src/views/v2/path/path_item_cache.rs

```diff
@@ -0,0 +1,258 @@
+use crate::core::v2::pathfinding::PathfindingResult;
+use crate::error::Result;
+use crate::modelsv2::TrackSectionModel;
+use crate::RetrieveBatchUnchecked;
+use editoast_schemas::infra::TrackOffset;
+use std::collections::HashMap;
+use std::collections::HashSet;
+
+use editoast_models::DbConnection;
+use editoast_schemas::train_schedule::PathItemLocation;
+
+use crate::modelsv2::OperationalPointModel;
+
+type TrackOffsetResult = std::result::Result<Vec<Vec<TrackOffset>>, TrackOffsetExtractionError>;
+
+/// Gather information about several path items, factorizing db calls.
+pub struct PathItemCache {
+    uic_to_ops: HashMap<i64, Vec<OperationalPointModel>>,
+    trigram_to_ops: HashMap<String, Vec<OperationalPointModel>>,
+    ids_to_ops: HashMap<String, OperationalPointModel>,
+    valid_tracks: HashSet<String>,
```

Comment:
This could also be renamed
```suggestion
    existing_track_ids: HashSet<String>,
```